### PR TITLE
Documentation for the trigger channels

### PR DESCRIPTION
**Pull request by phxql**
Hi,

this PR adds the documentation for the trigger channels (which are implemented in https://github.com/eclipse/smarthome/pull/1936)

### CODE REVIEW

File: docs/documentation/development/bindings/thing-handler.md [hunk 1]
```diff
@@ -125,6 +125,22 @@ public void dispose() {
...
+triggerChannel("channelId")
+```
+
+If an event payload is needed, use the overloaded version:
+
+```java
+triggerChannel("channelId", "PRESSED")
+```
+
+The call will be delegated to the framewort. It is binding specific when the channel should be triggered.
```
Comment (solderra) [inline docs/documentation/development/bindings/thing-handler.md]: typo: should be "framework"

File: docs/documentation/features/dsl.md [hunk 2]
```diff
@@ -62,6 +62,22 @@ The referenced bridge is specified in the parentheses. Please notice that the UI
 
 It is also possible to manually define channels in the DSL. Usually this is not needed, as channels will be automatically created by the binding based on the thing type description. But there might be some bindings, that require the manual definition of channels.
 
+### State channels
+
+```
+Thing yahooweather:weather:losangeles [ location="2442047", unit="us", refresh=120 ] {
```
Comment (kaikreuzer) [inline docs/documentation/features/dsl.md]: Afair, the location must be specified without quotes by now.

File: docs/documentation/features/dsl.md [hunk 3]
```diff
@@ -72,4 +88,15 @@ Thing yahooweather:weather:losangeles [ location="2442047", unit="us", refresh=1
 }
 ```
 
-Each channel definition must be placed inside the curly braces and begin with the accepted item type (e.g. String). After this the channel ID follows with the configuration of a channel. The framework will merge the list of channels coming from the binding and the user-defined list in the DSL.
+### Trigger channels
+
+```
+Thing yahooweather:weather:losangeles [ location="2442047", unit="us", refresh=120 ] {
```
Comment (kaikreuzer) [inline docs/documentation/features/dsl.md]: Afair, the location must be specified without quotes by now.

File: docs/documentation/development/bindings/thing-handler.md (hunk 1 above)
Comment (phxql) [inline docs/documentation/development/bindings/thing-handler.md]: Fixed.

File: docs/documentation/features/dsl.md (hunk 2 above)
Comment (phxql) [inline docs/documentation/features/dsl.md]: Fixed all locations. They are decimal values in the binding xml:

```xml
            <parameter name="location" type="integer" required="true">
```

File: docs/documentation/features/dsl.md (hunk 3 above)
Comment (phxql) [inline docs/documentation/features/dsl.md]: See above.

**Comment by kaikreuzer** [comment]
As the feature has been merged, we can also merge the docs for it - many thanks!

**Pull request merged**